src/lib/paste.ts

```diff
@@ -39,7 +48,46 @@ const removeIDs = (slice: ManuscriptSlice) => {
   })
 }
 
-export const transformPasted = (slice: ManuscriptSlice): ManuscriptSlice => {
+const pasteSection = (slice: ManuscriptSlice, view: ManuscriptEditorView) => {
+  const { state, dispatch } = view
+  let sectionTitleNode: ManuscriptNode | null = null
+  const sectionBodyNodes: ManuscriptNode[] = []
+  ClipboardEvent
+
+  slice.content.descendants((node) => {
+    if (node.type === node.type.schema.nodes.section) {
+      node.content.forEach((item) => {
+        const sectionBodyNode = state.schema.nodes.paragraph.create(
+          {},
+          item.content
+        ) as ParagraphNode
+
+        sectionBodyNodes.push(sectionBodyNode)
+      })
+    } else if (
+      node.type === node.type.schema.nodes.section_title &&
+      node.content.childCount === 1
+    ) {
+      sectionTitleNode = state.schema.nodes.section_title.create(
+        {},
+        node.content
+      ) as SectionTitleNode
+    }
+  })
+  if (sectionTitleNode) {
+    const pastedSectionContents = [sectionTitleNode, ...sectionBodyNodes]
+    insertSection(false, pastedSectionContents)(state, dispatch, view)
+    event?.preventDefault() // stop other pasting handlers
+    event?.stopPropagation()
```

Comment:
we can move the selection to the end of the section and wrap the slice with the section, without interfering with the event.

Reply:
I tried this, but unfortunately, it only worked for sections with multiple paragraphs, as we discussed earlier. It didn't handle sections with a single paragraph or element, so I had to adjust the approach. Please let me know if you have any other ideas.